types/webpack-dev-middleware/index.d.ts

```diff
@@ -66,7 +66,7 @@ declare namespace WebpackDevMiddleware {
         callbacks: Callback[];
         options: Options;
         compiler: webpack.Compiler;
-        watching: webpack.Watching | null;
+        watching: webpack.Compiler["watching"] | null;
```

Comment:
With the latest version of Webpack this is not needed anymore, `webpack.Watching` is now available.